File: apps/desktop/src/pastePayload.ts
/** Mirror of `tether_core::pty_input::paste_payload` for the desktop paste path. */

export const PASTE_START = '\u001b[200~';
export const PASTE_END = '\u001b[201~';

/**
 * Bytes to send for a paste, fenced when the program has bracketed paste on.
 *
 * The clipboard is untrusted. Text carrying its own `ESC[201~` would close the
 * fence early, and everything after it would arrive as typing — the next
 * newline then runs as Enter. Strip markers whether or not the fence goes on.
 */
export function pastePayload(text: string, bracketed: boolean): string {
  const clean = text.split(PASTE_START).join('').split(PASTE_END).join('');
  if (bracketed) return `${PASTE_START}${clean}${PASTE_END}`;
  return clean;
}

File: clients/apple/TetherKit/Sources/TetherKit/Terminal/PastePayload.swift
/// A paste fenced in `ESC[200~`/`ESC[201~` when the program asked (DECSET 2004).
///
/// Markers inside the clipboard are stripped either way: a pasted `ESC[201~`
/// would end the fence early and run the rest as typed commands.
enum PastePayload {
  static let start = "\u{1B}[200~"
  static let end = "\u{1B}[201~"

  static func make(_ text: String, bracketed: Bool) -> String {
    let clean = text.replacingOccurrences(of: start, with: "").replacingOccurrences(of: end, with: "")
    return bracketed ? start + clean + end : clean
  }
}
